Lib/zipfile/__init__.py

```diff
@@ -896,34 +889,20 @@ def __init__(self, fileobj, mode, zipinfo, pwd=None,
 
         if hasattr(zipinfo, 'CRC'):
             self._expected_crc = zipinfo.CRC
-            self._running_crc = crc32(b'')
+            self._orig_start_crc = crc32(b'')
         else:
             self._expected_crc = None
+            self._orig_start_crc = None
 
         self._seekable = False
         try:
             if fileobj.seekable():
                 self._orig_compress_start = fileobj.tell()
-                self._orig_compress_size = zipinfo.compress_size
                 self._orig_file_size = zipinfo.file_size
-                self._orig_start_crc = self._running_crc
-                self._orig_crc = self._expected_crc
                 self._seekable = True
         except AttributeError:
             pass
-
-        self._decrypter = None
-        if pwd:
-            if zipinfo.flag_bits & _MASK_USE_DATA_DESCRIPTOR:
-                # compare against the file type from extended local headers
-                check_byte = (zipinfo._raw_time >> 8) & 0xff
-            else:
-                # compare against the CRC otherwise
-                check_byte = (zipinfo.CRC >> 24) & 0xff
-            h = self._init_decrypter()
-            if h != check_byte:
-                raise RuntimeError("Bad password for file %r" % zipinfo.orig_filename)
-
+        self._init_read()
```

Comment:
"init" implies there is some one time setup/initialization going on to me. I'd suggest maybe `_reset_read_state`
```suggestion
        self._reset_read_state()
```